src/+replab/+equivariant/ForCompactGroup.m

```diff
@@ -2,22 +2,103 @@
+            nSamplesPerIter = 3; % number of samples per iteration
+            useInverses = true; % use inverses of samples as well
+            maxIters = 5000; % maximum number of iterations before erroring
+            nWarmUpIters = 20; % warm up iterations before first fit
+            nCheckIters = 10; % iterations to verify the noise floor
+            maxNoiseFloorError = 1; % maximum error on the noise floor (in log10)
+            noiseMargin = 1; % noise floor margin (in log10)
+            iter = 1;
+            errs = [];
+            % we fit 10^logfloor + exp(-slope*x + offset)
+            modelfun = @(b,x) log10(10^b(1)+10.^(-b(2)*x+b(3)));
+
+            for iter = 1:maxIters
+                acc = zeros(self.repR.dimension, self.repC.dimension);
+                for j = 1:nSamplesPerIter
                     g = self.group.sample;
-                    S = S + self.repR.matrixRowAction(g, self.repC.matrixColAction(g, X));
+                    S1 = self.repR.matrixRowAction(g, self.repC.matrixColAction(g, X));
+                    if useInverses
+                        ginv = self.group.inverse(g);
+                        S2 = self.repR.matrixRowAction(ginv, self.repC.matrixColAction(ginv, X));
+                        acc = acc + S1 + S2;
+                    else
+                        acc = acc + S1;
+                    end
+                end
+                if useInverses
+                    X1 = acc/(2*nSamplesPerIter);
+                    d = norm(X1 - X, 'fro');
+                    X = X1;
+                else
+                    X1 = acc/nSamplesPerIter;
+                    d = norm(X1 - X, 'fro');
+                    X = X1;
+                end
+                errs(1, iter) = d + 1e-100;
+                if iter > nWarmUpIters
+                    logfloor0 = log10(min(errs));
+                    slope0 = (log10(errs(1)) - log10(errs(nWarmUpIters)))/nWarmUpIters;
+                    offset0 = log10(errs(1));
+                    beta0 = [logfloor0 slope0 offset0];
+                    w = warning('off');
+                    errored = false;
+                    try
+                        [beta, R, J, CovB, MSE] = nlinfit(1:iter, log10(errs(1:iter)), modelfun, beta0);
+                    catch
+                        errored = true;
+                    end
+                    warning(w);
+                    if errored || rank(J) < 3
+                        % the rank is deficient, usually because the curve didn't flatten
+                        % yet, and thus the noise floor cannot be estimated
+                    else
+                        logfloor = beta(1);
+                        slope = beta(2);
+                        offset = beta(3);
+                        % compute error on fit parameters
+                        delta = sqrt(diag(CovB)) * tinv(1-0.05/2, iter - 3);
+                        crossing = [];
+                        if delta < maxNoiseFloorError/2
```

Comment:
I don't understand this line: `delta` is 3x1 and `maxNoiseFloorError/2` is 1x1. Do we want all three values of delta to be smaller than `maxNoiseFloorError/2` (than maybe using `max(delta)` would be clearer), or just a specific component?

Reply:
Indeed, that's only the first component which represents the error on the estimation noise floor. That has been corrected.

Reply:
Good, thanks!